proxy/pkg/proxy/proxy.go

```diff
@@ -37,6 +38,12 @@ func NewMultiHostReverseProxy(opts ...Option) *MultiHostReverseProxy {
 	}
 	rp.Director = rp.directorSelectionDirector
 
+	rp.Transport = &http.Transport{
```

Comment:
Could you check that the timeouts of the transport are not reset to 0 and thus get disabled? This used to be a source of fd leaks in the past.

https://golang.org/src/net/http/transport.go

Reply:
You can copy the DefaultTransport and override that. (see transport.go)